### PR TITLE
Confirm changes conversion tracking

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #8821 

## Relevant technical choices

* Diverging from IB: Excluded unnecessary `useCallback` to wrap action dispatch
* Change on the save button label requires additional check for conversion tracking settings change, but `Footer` component has already complexity of `20`, additional check goes to `21`. So I refactored it a bit, to extract the 2 components from inline components with a lot of conditional checks, into individual components.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by benbowler — CHANGES_REQUESTED** [review]
Hey @zutigrm, the code here looks good, however it does not address one of the key elements from the AC:

> When the state of the "Enable enhanced conversion tracking" toggle is changed in the edit view of either Ads or GA4 settings edit views, the "Save" CTA button should be re-labelled to read as "Confirm changes"

The button copy does change for other field changes in the Ads and Analytics settings components.

**Comment by zutigrm** [comment]
@benbowler Thanks, got the requirement mixed up with 8618 issue it was fixing. PR updated.

Side note - change on the label requires additional check for conversion settings change, but `Footer` component has already complexity `20`, additional check goes to `21`. So I refactored it a bit, to extract the 2 components from inline components with a lot of conditional checks, into individual components.

**Review by benbowler — APPROVED** [review]
Thanks @zutigrm, looking good. Moving to MR.

**Comment by zutigrm** [comment]
@eugene-manuilov It seems your review isn't saved when you sent it back to Execution. Can you please check? Thanks

**Review by eugene-manuilov — COMMENTED** [review]
Thanks, @zutigrm, but this is not how this issue should be fixed. We need to revert your changes and update the `rollbackChanges` functions in the settings datastore for both Analytics-4 and Ads modules to rollback conversion tracking changes:
https://github.com/google/site-kit-wp/blob/e9c3f4ab005a9f538cf214cc70e2db9dd7cbf140/assets/js/modules/analytics-4/datastore/settings.js#L236-L242

Plus, we need to update the `createSettingsStore` factory to allow overriding the default `haveSettingsChanged` selector similarly to how we do it for `createSubmitChangesStore` in the `createModuleStore` factory. This will require creating a new `makeDefaultHaveSettingsChanged` function similar to what we have for `makeDefaultCanSubmitChanges`.

File: assets/js/googlesitekit/data/create-settings-store.js [hunk 1]
```diff
@@ -80,6 +82,7 @@ export const createSettingsStore = (
 		storeName = undefined,
 		settingSlugs = [],
 		initialSettings = undefined,
+		haveSettingsChanged = null,
```
Comment (eugene-manuilov) [inline assets/js/googlesitekit/data/create-settings-store.js:85]: Let's call it `validateHaveSettingsChanged` to be consistent with other overrides.

File: assets/js/googlesitekit/data/create-settings-store.js [hunk 2]
```diff
@@ -280,23 +283,25 @@ export const createSettingsStore = (
...
+				return makeDefaultHaveSettingsChanged( {
+					state,
+					keys,
+					select,
+				} );
 			}
-
-			return ! isEqual( settings, savedSettings );
-		},
+		),
```
Comment (eugene-manuilov) [inline assets/js/googlesitekit/data/create-settings-store.js:304]: Let's use the same approach as the `createValidationSelector` function uses. It should be something like this:

```js
haveSettingsChanged: createRegistrySelector( ( select ) => ( state, ...args ) => validateHaveSettingsChanged( select, state, ...args ) );
```

File: assets/js/googlesitekit/data/create-settings-store.js [hunk 3]
```diff
@@ -499,3 +504,25 @@ export function makeDefaultCanSubmitChanges( storeName ) {
...
+function makeDefaultHaveSettingsChanged( params ) {
+	const { state, keys } = params;
+	const { settings, savedSettings } = state;
+
+	if ( keys ) {
+		return ! isEqual( pick( settings, keys ), pick( savedSettings, keys ) );
+	}
+
+	return ! isEqual( settings, savedSettings );
+}
```
Comment (eugene-manuilov) [inline assets/js/googlesitekit/data/create-settings-store.js:528]: This function should be similar to `makeDefaultCanSubmitChanges`:
```js
function makeDefaultHaveSettingsChanged() {
    return ( select, state, keys ) {
        ...
    }
}
```

File: assets/js/googlesitekit/modules/create-module-store.js [hunk 4]
```diff
@@ -72,6 +72,7 @@ export function createModuleStore( slug, args = {} ) {
 		requiresSetup = true,
 		submitChanges,
 		rollbackChanges,
+		haveSettingsChanged = null,
```
Comment (eugene-manuilov) [inline assets/js/googlesitekit/modules/create-module-store.js:75]: Let's rename this to be `validateHaveSettingsChanged`.

File: assets/js/googlesitekit/modules/create-module-store.js [hunk 5]
```diff
@@ -118,6 +119,7 @@ export function createModuleStore( slug, args = {} ) {
 				storeName,
 				settingSlugs,
 				initialSettings,
+				haveSettingsChanged,
```
Comment (eugene-manuilov) [inline assets/js/googlesitekit/modules/create-module-store.js:122]: And after all renamings and updates this should be:
```js
validateHaveSettingsChanged: validateHaveSettingsChanged || makeDefaultHaveSettingsChanged(),
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @zutigrm. Almost looks good to be, but let's update it to be more consistent with other `make*` and `validate*` functions that we use there. Please, see my comments.

**Review by eugene-manuilov — APPROVED** [review]
Great! Thanks, @zutigrm.

**Pull request merged**